duckduckgo/macos-browser
# Manage Private Email Addresses

Task/Issue URL: https://app.asana.com/0/72649045549333/1204615490114675/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/396

**Description**:

**Steps to test this PR**:


Address Management
---
1. Enable Email protection, if not enabled Already.  ( ··· > Email Protection - > Enable Email Protection)
2. Generate a Private Email Address (··· > Email Protection > Generate Private Email Address)
3. Open the Autofill UI, and create a login using the newly created Email address as username
4. 👁️ Observe the new Management UI Shows Up as follows:

<img width="525" alt="Screenshot 2023-07-11 at 11 35 55 AM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/23983e2c-40df-4c6e-8020-c490933573c0">

5. Hit the "Toggle"
6. 👁️ Observe the address is deactivated, and the UI is updated as follows:

<img width="525" alt="Screenshot 2023-07-11 at 11 36 20 AM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/b7d448a8-f693-455e-bd94-4636618aa823">

8. Activate the address
9. Create a new login item
10. Enter a non-existent address ending with @duck.com
11. 👁️ Observe the newly created address doesnot show Activate/deactivate controls
12. Close the Autofill management UI
13. Disable Wifi
14. Open the Autofill Management UI and select your valid @duck.com address
15. 👁️ Observe an error appears as follows:  (It'll take 5 secons to update due to timeout)

<img width="527" alt="Screenshot 2023-07-11 at 11 37 06 AM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/cc2b7a8d-c8d0-431a-ad6b-0c9dfa80db28">


Partial account updating
---
1. Go to [fill.dev email registration](https://fill.dev/form/registration-email)
2. Auto-generate **a password** in the login/signup form

<img width="478" alt="Screenshot 2023-07-03 at 8 06 58 PM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/2fe38681-21a0-4494-8e73-ba11ff2603c1">

3. 👁️ Open the login management UI and observe a new item was created with the generated password and an empty username
4. Generate a private email from the form

<img width="433" alt="Screenshot 2023-07-03 at 8 07 14 PM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/12e98287-2e67-413e-9c48-6c9c445fd432">

5. 👁️ Observe the previously created item includes both username and password
6. Submit the form
7. 👁️ Observe a "New login saved" popup appears correctly

<img width="350" alt="Screenshot 2023-07-03 at 8 07 27 PM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/c6fe7efe-bfd6-4370-a995-34600b8aa28b">

Partial account updating (Various updates)
---

1. Go to [fill.dev email registration](https://fill.dev/form/registration-email)
2. Auto-generate **a private email** in the login/signup form
3. 👁️ Open the login management UI and observe a new item was created with the generated email and an empty password
4. Generate a password
5. 👁️ Observe the previously created item includes both username and password
6. Generate a new email using the form
7. 👁️ Observe the previously created item includes the updated email
8. Generate a new password
9. 👁️ Observe the previously created item includes the updated password 


Credentials save prompt
---
1. Go to [netflix.com](https://www.netflix.com/es-en/login)
2. Type 'testing123' as the username
3. Type '1234567890' as the password
4. Submit the form
5. 👁️ Observe you get a "Save login?" dialog and it's CTA button says "Save"
6. Hit "Save"
7. 👁️  Observe a new login item was created and username/password are correct

<img width="349" alt="Screenshot 2023-07-03 at 8 17 00 PM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/8a0bcbb9-1270-41b3-b495-16fd80f985b1">

Credentials update prompt
---
1. Close the browser and reopen
2. Go to [fill.dev user registration](https://fill.dev/form/registration-username)
3. Type 'testing123' as the username (or the same username you've used in the previous section)
4. Type '0987654321' as the password (or any other password)
5. Submit the form
6. 👁️ Observe you get a "Save login?" dialog, but in this case 
8. Hit "Update"
9. 👁️ Observe the login item was updated and username/password are correct

<img width="357" alt="Screenshot 2023-07-03 at 8 17 20 PM" src="https://github.com/duckduckgo/macos-browser/assets/1156669/1b8381c3-5b14-4640-b666-dc0f303352ee">



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
